File: 1844-replace-all-digits-with-characters/1844-replace-all-digits-with-characters.java
class Solution {
    public String replaceDigits(String s) {
        String res = "";
        
        for(int i = 0; i < s.length(); i++){
            if(i % 2 != 0){
                char shift = (char) ((int) s.charAt(i-1) + Character.getNumericValue(s.charAt(i)));
                res += shift;
            } else {
                res += s.charAt(i);
            }
        }
        
        return res;
    }
}

File: 1844-replace-all-digits-with-characters/1844-replace-all-digits-with-characters.py
class Solution:
    def replaceDigits(self, s: str) -> str:
        res = ""
        
        for i in range(len(s)):
            if i % 2 != 0:
                b = chr(ord(s[i-1])+ int(s[i]))
                res += b
            else:
                res += s[i]
        
        return res